content/chapters/compute/lab/content/processes.md

```diff
@@ -114,17 +114,82 @@ From the outside, it's as if you were running these commands from the terminal.
 Head over to `support/sleepy/sleepy_creator.py`.
 Use `subprocess.Popen()` to spawn 10 `sleep 1000` processes.
 
-1. Now use the same `pstree -pac` command and look for `sleepy_creator.py`.
-It is a `python3` process, as this is the interpreter that runs the script, but we call it the `sleepy_creator.py` process for simplicity.
-If you found it, you did something wrong.
-It should be missing.
-Now use `pstree -pac` and look for the `sleep` processes you have just created.
-
-[Quiz](../quiz/parent-of-sleep-processes.md)
-
-1. Change the code in `sleepy_creator.py` so that the `sleep 1000` processes are the children of `sleepy_creator.py`.
-Kill the previously created `sleep` processes using `killall sleep`.
-Verify that `sleepy_creator.py` remains the parent of the `sleep`s it creates using `pstree -pac`.
+1. Solve `TODO 1`: use `subprocess.Popen()` to spawn 10 `sleep 1000` processes.
+
+   Start the script:
+
+   ```console
+   student@os:~/.../lab/support/sleepy$ python3 sleepy_creator.py
+   ```
+
+   Look for the parent process:
+
+   ```console
+   student@os:~$ ps -e -H -o pid,ppid,cmd | (head -1; grep "python3 sleepy_creator.py")
+   ```
+
+   It is a `python3` process, as this is the interpreter that runs the script, but we call it the `sleepy_creator.py` process for simplicity.
+   No output will be provided by the above command, as the parent process (`sleepy_creator.py`) dies before its child processes (the 10 `sleep 1000` subprocesses) finish their execution.
+   The parent process of the newly created child processes is an `init`-like process: either `systemd`/`init` or another system process that adopts orphan processes.
+   Look for the `sleep` child processes using:
+
+   ```console
+   student@os:~$ ps -e -H -o pid,ppid,cmd | (head -1; grep sleep)
+    PID    PPID         CMD
+   4164    1680     sleep 1000
+   4165    1680     sleep 1000
+   4166    1680     sleep 1000
+   4167    1680     sleep 1000
+   4168    1680     sleep 1000
+   4169    1680     sleep 1000
+   4170    1680     sleep 1000
+   4171    1680     sleep 1000
+   4172    1680     sleep 1000
+   4173    1680     sleep 1000
+   ```
+
+   Notice that the child processes do not have `sleepy_creator.py` as a parent.
+   What's more, as you saw above, `sleepy_creator.py` doesn't even exist anymore.
+   The child processes have been adopted by an `init`-like process (in the output above that process has PID `1680` - `PPID` stands for _parent process ID_).
+
+   [Quiz](../quiz/parent-of-sleep-processes.md)
+
+1. Change the code in `sleepy_creator.py` so that the `sleep 1000` processes remain the children of `sleepy_creator.py`.
```

Comment:
```suggestion
1. Solve `TODO 2`: change the code in `sleepy_creator.py` so that the `sleep 1000` processes remain the children of `sleepy_creator.py`.
```